### PR TITLE
Add query MR EFA direct verb

### PR DESCRIPTION
**Pull request by mrgolin**
This PR is adding a new EFA DV interface to query for MR attributes, with an initial implementation that returns PCIe bus IDs used by the device to reach the MR to write data for receive, RDMA read, and RDMA write receive operations.
Documentation was updated accordingly and pyverbs layer and a test to cover this new functionality are being added.

### CODE REVIEW

File: providers/efa/man/efadv_query_mr.3.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+struct efadv_mr_attr {
+	uint64_t comp_mask;
+	uint16_t pci_bus_id_validity;
+	uint16_t recv_pci_bus_id;
+	uint16_t rdma_read_pci_bus_id;
+	uint16_t rdma_recv_pci_bus_id;
+};
+```
+
+*inlen*
```
Comment (shijin-aws) [inline providers/efa/man/efadv_query_mr.3.md:39]: is `inlen` part of the struct? Didn't see it

Comment (shijin-aws) [reply providers/efa/man/efadv_query_mr.3.md:39]: Nvm it's an argument in `efadv_query_mr` call.

**Comment by mrgolin** [comment]
Fixed some Cython 0.26 version specific compilation error and now all tests have passed.

**Comment by shijin-aws** [comment]
Can we have someone review this PR? We have a Libfabric PR pending on this https://github.com/ofiwg/libfabric/pull/9525

**Comment by mrgolin** [comment]
Updated naming for consistency with the kernel patch changes.

File: providers/efa/verbs.c [hunk 2]
```diff
@@ -279,6 +279,62 @@ struct ibv_mr *efa_reg_mr(struct ibv_pd *ibvpd, void *sva, size_t len,
...
+	fill_attr_out(cmd, EFA_IB_ATTR_QUERY_MR_RESP_RDMA_RECV_IC_ID,
+		      &rdma_recv_ic_id, sizeof(rdma_recv_ic_id));
+
+	err = execute_ioctl(ibvmr->context, cmd);
+	if (err) {
+		verbs_err(verbs_get_ctx(ibvmr->context), "Failed to query MR\n");
+		return err;
+	}
+
+	if (ic_id_validity & EFA_QUERY_MR_VALIDITY_RECV_IC_ID) {
```
Comment (jgunthorpe) [inline providers/efa/verbs.c:322]: This ic_id_validity thing doesn't make alot of sense, we already have UVERBS_ATTR_F_VALID_OUTPUT to signal when the attribute was supported. I'd suggest just keeping this simple, make the interconnect ID a u64 and have -1ULL mean not supported, then you can pre-init it to -1 and not worry about this.

**Comment by mrgolin** [comment]
@rleon @jgunthorpe Can you please merge this? We need it in the upcoming release.

**Comment by rleon** [comment]
> @rleon @jgunthorpe Can you please merge this? We need it in the upcoming release.

At least you need to update first patch to right kernel commit.

Thanks

**Comment by mrgolin** [comment]
Updated kernel headers. Thanks.

**Pull request merged**